File: backend/model/ref.go
package model

type Ref struct {
	ID          int64  `db:"id" json:"id"`
	Title       string `db:"title" json:"title"`
	Description string `db:"description" json:"description"`
	Link        string `db:"link" json:"link"`
	UserID      int64  `db:"user_id" json:"user_id"`
	NoteID      int64  `db:"note_id" json:"note_id"`
}

File: frontend/src/components/Ref/Ref.js
import React from "react";
import { Card, Icon, Modal } from "antd";

import RefInput from "../RefInput";
import "./Ref.css";

const Ref = ({ id, title, description, link, onClickEdit, onClickDelete }) => (
  <Card
    className="Ref"
    type="inner"
    title={title}
    extra={
      <div className="Ref-icons">
        <Icon type="edit" onClick={onClickEdit} />
        <Icon type="delete" onClick={() => onClickDelete(id)} />
      </div>
    }
  >
    <a target="_blank" rel="noopener noreferrer" href={link}>
      {link}
    </a>
    <p>{description}</p>
  </Card>
);

export default Ref;
